src/client/render/pipeline.h

```diff
@@ -183,10 +183,10 @@ class TextureBufferOutput : public RenderTarget
 	TextureBufferOutput(TextureBuffer *buffer, const std::vector<u8> &texture_map, u8 depth_stencil);
 	virtual ~TextureBufferOutput() override;
 	void activate(PipelineContext &context) override;
+	TextureBuffer *buffer;
```

Comment:
Seems a bit ugly to expose this raw pointer publicly. Is it actually needed? In the code in core.cpp, the caller passes in the buffer and then uses this pointer to access it again, but they don't need to access it via this pointer I think?